compiler/noirc_evaluator/src/ssa/ssa_gen/context.rs

```diff
@@ -355,7 +356,10 @@ impl<'a> FunctionContext<'a> {
 
     /// Insert constraints ensuring that the operation does not overflow the bit size of the result
     ///
-    /// If the result is unsigned, overflow will be checked during acir-gen (cf. issue #4456), except for bit-shifts, because we will convert them to field multiplication
+    /// If the result is unsigned, overflow will be checked during acir-gen (cf. issue #4456), except for:
+    /// - bit-shifts, because we will convert them to field multiplication
+    /// - when multiplying two u128: the overflow will be checking during acir-gen, but we also need to do an
+    ///   extra check because the result might wrap around the field maximum value and end up in a valid u128 value
```

Comment:
```suggestion
    /// - when multiplying two u128: the overflow will be checked during acir-gen, but we also need to do an
    ///   extra check because the result might wrap around the field maximum value and end up in a valid u128 value
```